internals-js/src/coreSpec.ts

```diff
@@ -507,18 +507,16 @@ export class FeatureDefinitions<T extends FeatureDefinition = FeatureDefinition>
       return this;
     }
     this._definitions.push(definition);
-    // We sort by decreased versions (this makes `find` a bit easier, and it feels somewhat natural anyway to have more
-    // recent versions first).
+    // We sort by decreased versions sa it feels somewhat natural anyway to have more recent versions first.
     this._definitions.sort((def1, def2) => -def1.version.compareTo(def2.version));
     return this;
   }
 
   /**
-   * Returns the known definition with the greatest version that satisfies the requested version, or undefined if no
-   * known version can satisfy this version.
+   * Returns the definition corresponding to the requested version if known.
    */
   find(requested: FeatureVersion): T | undefined {
-    return this._definitions.find(def => def.version.satisfies(requested));
```

Comment:
Note that this is the main fix here: because 2.2 "satisfies" 2.1 and 2.0, it was picked up every time. But in hindsight, if the user manually specify a given version in the subgraph, no reason to ever give it anything else, so juste made it an equal.

Note that it's possible that some core feature processor would want a method like the one that existed, which gives you whatever most recent version is able to "read" the requested version, but we can add a separate method for that if we ever need it. What we want to now is "expanding" the definitions for a version, so an equal match makes more sense.

Reply:
There's no further traversal that would need to happen and would just be an exact equality in all cases, then?

Reply:
There is still a traversal since we keep the definition as a sorted list and not a map keyed by version (javascript is a bit wanky when it goes to make keyed by something else than an base type), but the match is on exactly equality in all cases so we'll never expand to a version that wasn't the exact one asked (worst case, the version is unknown and we error saying that, but that's kind of what we want anyway).

Reply:
Awesome, thank you for the explanation.  Looks good to me.